tools/protodoc/protodoc.py

```diff
@@ -388,7 +401,30 @@ def FormatAnchor(label):
   return '.. _%s:\n\n' % label
 
 
-def FormatFieldAsDefinitionListItem(outer_type_context, type_context, field):
+def FormatSecurityOptions(security_option, field, type_context, edge_default_yaml):
+  sections = []
+
+  if security_option.configure_for_untrusted_downstream:
+    sections.append(
+        Indent(4, 'This field should be configured in the presence of untrusted *downstreams*'))
+  if security_option.configure_for_untrusted_upstream:
+    sections.append(
+        Indent(4, 'This field should be configured in the presence of untrusted *upstreams*'))
+
+  validate_fragment.ValidateFragment(field.type_name[1:], edge_default_yaml)
+  field_name = type_context.name.split('.')[-1]
+  example = {field_name: edge_default_yaml}
+  sections.append(
+      Indent(4, 'Example configuration for untrusted environments:\n\n') +
+      Indent(4, '.. code-block:: yaml\n\n') +
+      '\n'.join(IndentLines(6,
+                            yaml.dump(example).split('\n'))))
+
+  return '.. note::\n' + '\n\n'.join(sections)
```

Comment:
Thoughts on attention vs. note? I could go either way but security seems like an attention type of thing.